core/base/src/TDirectory.cxx

```diff
@@ -1028,9 +1028,12 @@ void TDirectory::FillFullPath(TString& buf) const
 ///    gDirectory->cd("b");
 ///    gDirectory->mkdir("d");
 /// ~~~
+/// ifNotExist is just for compatibility with the TDirectoryFile override
 
-TDirectory *TDirectory::mkdir(const char *name, const char *title)
+TDirectory *TDirectory::mkdir(const char *name, const char *title, Bool_t ifNotExist)
 {
+   if (ifNotExist)
+      Error("mkdir", "ifNotExist only valid for TDirectoryFile mkdir", name);
```

Comment:
@pcanal good [question](https://github.com/root-project/root/pull/4649#discussion_r350855767). The `TDirectory` implementation does not check whether the directory already exists and simply allows you to make another directory with the same name. I'm not sure the reasoning behind this different implementation, but there didn't seem to be a need for `ifNotExist`.

```python
In [1]: import ROOT

In [2]: d = ROOT.TDirectory('motherdir', 'motherdir')

In [3]: sd = d.mkdir('daughterdir')

In [4]: sd2 = d.mkdir('daughterdir')

In [5]: sd
Out[5]: <ROOT.TDirectory object ("daughterdir") at 0x7fa487cf9c90>

In [6]: sd2
Out[6]: <ROOT.TDirectory object ("daughterdir") at 0x7fa487b09a90>

In [7]: sd is sd2
Out[7]: False
```

Reply:
> Looks good. Thanks. One last question (my apologies if I missed the answer): why a difference in behavior between TDirectory and TDirectoryFile?

Sorry, @pcanal, I neglected to submit this review. Answer above.